### PR TITLE
Visualization JSON generator added

### PR DESCRIPTION
**Pull request by manalishah**

### CODE REVIEW

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/Labels.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package edu.usc.ir.visualization;
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/Labels.java:17]: Need ALV2 license header

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+     	nltk = new HashMap<String,Integer>();
+  		nlp = new HashMap<String,Integer>();
+  		tika = null;
+  		
+//  	Enable NLTK Named Entity Parser
+  		System.setProperty(NamedEntityParser.SYS_PROP_NER_IMPL, NLTKNERecogniser.class.getName());
+  		try {
+			tika = new Tika(new TikaConfig(NLTKandCoreNLP.class.getResourceAsStream("tika-config.xml")));
+		} catch (TikaException | IOException | SAXException e) {
+			// TODO Auto-generated catch block
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    
+    private Tika tika;
+    private Metadata md;
+    public NLTKandCoreNLP(){
+    	freq = new HashSet<String>();
+     	nltk = new HashMap<String,Integer>();
+  		nlp = new HashMap<String,Integer>();
+  		tika = null;
+  		
+//  	Enable NLTK Named Entity Parser
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+            		   .create(url, username, password, null)
+            		   .accept(MediaType.APPLICATION_JSON)
+            		   .get()
+            		   .readEntity(String.class);
+
+            try {
+                node = mapper.readTree(response);
+                dataset= node.get("response").get("docs");
+            } catch (JsonProcessingException e) {
+                // TODO Auto-generated catch block
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+                e.printStackTrace();
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+                        if(nlp.containsKey(names[i])){
+                            nlp.put(names[i], nlp.get(names[i]) + 1);
+                        }
+                        else{
+                            nlp.put(names[i], 1);
+                        }
+                    }
+                }
+                
+//                extract CoreNLP dates entity
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+                            nlp.put(names[i], nlp.get(names[i]) + 1);
+                        }
+                        else{
+                            nlp.put(names[i], 1);
+                        }
+                    }
+
+                }
+
+//                extract CoreNLP organizations entity
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 8]
```diff
@@ -0,0 +1,276 @@
...
+                String content = datasetElement.get("content").asText();
+                md = new Metadata();
+                try (InputStream stream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8))) {
+                    tika.parse(stream, md);
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+                
+//                extract CoreNLP locations entity
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+            Iterator<JsonNode> datasetElements = dataset.iterator();
+            
+            while (datasetElements.hasNext()) {
+                JsonNode datasetElement = datasetElements.next();
+                String content = datasetElement.get("content").asText();
+                md = new Metadata();
+                try (InputStream stream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8))) {
+                    tika.parse(stream, md);
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+                        }
+                        if(nltk.containsKey(ner_name)){
+                            nltk.put(ner_name, nltk.get(ner_name) + 1);
+                        }
+                        else{
+                            nltk.put(ner_name, 1);
+                        }
+                    }
+                }
+                //continue for each url in response.
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+            }
+            //fetch new set of urls
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 12]
```diff
@@ -0,0 +1,276 @@
...
+        }
+    }
+
+	private void createJSON(File destination) throws JsonGenerationException, JsonMappingException, IOException {
+		// TODO Auto-generated method stub
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+                nltk_value.add(0);
+            }
+            if(nlp.containsKey(value)){
+                nlp_value.add(nlp.get(value));
+            }
+            else{
+                nlp_value.add(0);
+            }
+        }
+        //Formating the required json for visualization in https://github.com/manalishah/Tika-NLTKvsCoreNLP.git
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/NLTKandCoreNLP.java]: remove

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/Names.java [hunk 14]
```diff
@@ -0,0 +1,10 @@
+package edu.usc.ir.visualization;
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/Names.java:17]: ALv2 header

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/Series.java [hunk 15]
```diff
@@ -0,0 +1,24 @@
+package edu.usc.ir.visualization;
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/Series.java:17]: ALv2 header

File: nltkrest-examples/src/main/resources/edu/usc/ir/visualization/tika-config.xml [hunk 16]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (chrismattmann) [inline nltkrest-examples/src/main/resources/edu/usc/ir/visualization/tika-config.xml:1]: ALv2 header

**Comment by chrismattmann** [comment]
please also add the HTML and initially generated JSON

File: nltkrest-examples/src/main/java/edu/usc/ir/visualization/Labels.java (hunk 1 above)
Comment (manalishah) [inline nltkrest-examples/src/main/java/edu/usc/ir/visualization/Labels.java:17]: done

**Comment by manalishah** [comment]
thanks for the review @chrismattmann
the changes have been updated. 

HTML and initially generated index displayed in this repo. https://github.com/manalishah/Tika-NLTKvsCoreNLP.git
Also available as a github.io page : http://manalishah.github.io/Tika-NLTKvsCoreNLP/

**Comment by chrismattmann** [comment]
great job! @manalishah

**Pull request merged**